webservices/src/main/java/com/javaschool/webservices/configuration/RabbitMQConfig.java

```diff
@@ -0,0 +1,53 @@
+package com.javaschool.webservices.configuration;
+
+import org.springframework.amqp.core.Binding;
+import org.springframework.amqp.core.BindingBuilder;
+import org.springframework.amqp.core.Exchange;
+import org.springframework.amqp.core.ExchangeBuilder;
+import org.springframework.amqp.core.Queue;
+import org.springframework.amqp.core.QueueBuilder;
+import org.springframework.amqp.core.TopicExchange;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@Configuration
+public class RabbitMQConfig {
+
+	private final String SHIPPING_QUEUE_NAME;
+
+	private final String SHIPPING_EXCHANGE;
+
+	private final String SHIPPING_ROUTING_KEY;
+
+	public RabbitMQConfig(@Value("${rabbitmq.shippingQueueName}") String shippingQueueName,
```

Comment:
using `@Value` like this can quickly become problematic. Let's say you change the entry name in the YAML/properties file, you would break the code each place where this value is being read. Use encapsulation instead, to avoid these problems, by isolating this value the client code will see it throw an abstraction, it would be type safe and if you change the entry name you would only need to modify it in one place or even not at all.

Reply:
I change the way to use properties, now I created a @ConfigurationProperties class for rabbit configuration.